brainfuck/bf.rs

```diff
@@ -136,9 +136,9 @@ struct Program {
 }
 
 impl Program {
-    fn new(code: &str) -> Self {
+    fn new(code: &[u8]) -> Self {
         Self {
-            ops: parse(&mut code.chars()),
+            ops: parse(&mut code.iter().copied()),
```

Comment:
Can we avoid copying by owning the `code`? It's not used anywhere else. Require more refactoring though, so it's minor.

Reply:
I tried to use `Vec`/`Iterator` and it became slower by 1 sec. Copying here is cheaper.

Reply:
Passing by reference, like:

```rust
fn parse<'a>(it: &mut impl Iterator<Item = &'a u8>) -> Box<[Op]> {
    // ...
}
```

Made no difference in speed (`mandel.b` runs in ~23.4 sec in my machine).